server/src/main/java/org/opensearch/ingest/ConfigurationUtils.java

```diff
@@ -512,7 +512,9 @@ public static Processor readProcessor(
         String type,
         Object config
     ) throws Exception {
-        if (config instanceof Map) {
+        if (config == null) {
```

Comment:
Could you please update the other overloaded versions of `readProcessor` (where applicable) to account for `null`?

Reply:
@reta, there are two versions of readProcessor, I've checked the usage of another one, it's been called directly when creating foreach processor, null check must be added to it, I've done that, thanks for your review.